### PR TITLE
 LTD-5382-fix-hawk-dbt-platform

### PR DESCRIPTION
**Pull request by depsiatwal**
When migration to DBT Platform we had issue with CloudFront rejecting the request 
when we send some content when we doing a GET request. 

Then additional we required the hawk parameters to be empty else the hawkheader is missing hence verifyhawk function fails.

### CODE REVIEW

File: core/client.py [hunk 1]
```diff
@@ -83,17 +83,22 @@ def verify_hawk_response(response, sender, stream=False):
 
 
 def perform_request(method, request, appended_address, data=None, stream=False):
-    data = data or {}
```
Comment (depsiatwal) [inline core/client.py:86]: required to removed this as an empty dict is being sent on GET requests on DBT platform

File: core/client.py [hunk 2]
```diff
@@ -83,17 +83,22 @@ def verify_hawk_response(response, sender, stream=False):
...
     session = request.requests_session  # provided by RequestsSessionMiddleware
     url = _build_absolute_uri(appended_address.replace(" ", "%20"))
 
+    if not data:
+        # If we not sending any data then the hawk header should be defined with no content and no header
+        content_type = ""
```
Comment (depsiatwal) [inline core/client.py:91]: we require an empty content and content_type when we not sending any data as per hawk spec.

**Comment by depsiatwal** [comment]
This has been smoke tested on dev2 and all appears to be working fine.

**Review by markj0hnst0n — APPROVED** [review]
Did a bit of v basic testing on dev 2.  Looks grand.

**Review by kevincarrogan — APPROVED** [review]

**Pull request merged**